content/fast/incorporate-performance-budgets-into-your-build-tools/codelab-setting-performance-budgets-with-webpack.md

```diff
@@ -0,0 +1,193 @@
+A [good rule](https://web.dev/fast/your-first-performance-budget#budget-for-quantity-based-metrics)
+of thumb is to deliver under 170 KB of compressed/minified
+[critical-path](https://developers.google.com/web/fundamentals/performance/critical-rendering-path/)
+resources.
+
+<div class="aside note">
+    Critical-path resources are the minimal set of resources required
+    for the browser to display the first screen's worth of content.
+</div>
+
+For this simple demo, try being even more conservative and set the budget to
+100 KB. In the configuration file, add the following:
+
+<pre class="prettyprint"> 
+    module.exports = { 
+        //... 
+            performance: { 
+            maxAssetSize: 100000, 
+            maxEntrypointSize: 100000, 
+            hints: 'warning' 
+        } 
+    };
+</pre>
+
+The new performance budget is set in **bytes**:
+
+- 100000 bytes for individual assets (maxAssetSize)
+
+- 100000 bytes for the entry-point (maxEntrypointSize)
+
+In this case, there’s only one bundle which also acts as the entry point.
+
+Possible values for **hints** are:
+
+1. **`warning`** (default): Displays a yellow warning message, but the build
+   passes. It’s best to use this in development environments.
+
+2. **`error`**: Displays a red error message, but the build still passes. This
+   setting is recommended for production builds.
+
+3. **`false`**: No warnings or errors are shown.
+
+<figure> 
+    <img src="webpack-error.png" alt="" class="screenshot"> 
+    <figcaption>Webpack performance hint "error" 🚨</figcaption> 
+</figure>
+
+## Optimize
+
+The purpose of a performance budget is to warn you about performance issues
+before they become too difficult to fix. There is always more than one way to
+build an app and some techniques will make for faster load times. (A lot of
+them are documented right here in [Optimizing your JavaScript](https://web.dev/fast#topic-Optimize-your-JavaScript). 🤓)
+
+Frameworks and libraries make developers’ lives easier, but end users don’t
+really care how apps are built, only that they’re functional and fast. If you
+find yourself going over the performance budget, it’s time to think about
+possible optimizations.
+
+In the real world, large client-side frameworks are usually hard to swap out,
+so it’s important to use them wisely. With a bit of research, you can often
+find smaller alternatives to popular libraries that do the job just as well
+([date-fns](https://date-fns.org/) is a good alternative for moment.js).
+Sometimes it makes more sense to not use a framework or library at all if it
+has a significant performance impact.
+
+Removing unused code a good way to optimize apps that include large third-party
+libraries. The process is explained in detail in [this guide](https://web.dev/fast/remove-unused-code)
+and here’s a quick way to rewrite the countdown code without using moment.js.
+
+In **app/components/Countdown.jsx** remove:
+
+<pre class="prettyprint"> 
+    <s> 
```

Comment:
The `pre` tag counts the indentation in the code snippet. So I think if you want to use the strikethrough you may want to indent the code all the way to the left, and put an `<s>` on each line.

![image](https://user-images.githubusercontent.com/1066253/52148123-9d829a80-261d-11e9-80a6-0646fca376a4.png)